### PR TITLE
[doc] Add setter for reader subscription name

### PR DESCRIPTION
**Pull request by Jennifer88huang**
Master Issue: #8787
Related PR: https://github.com/apache/pulsar/pull/8801

### Modifications

Add doc content on adding setter for reader subscription name.

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@Jennifer88huang thanks for your contribution. Need to add similar contents for C++ client as well?

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -675,6 +676,7 @@ String|`topicName`|Topic name. |None
 int|`receiverQueueSize`|Size of a consumer's receiver queue.<br/><br/>For example, the number of messages that can be accumulated by a consumer before an application calls `Receive`.<br/><br/>A value higher than the default value increases consumer throughput, though at the expense of more memory utilization.|1000
 ReaderListener&lt;T&gt;|`readerListener`|A listener that is called for message received.|None
 String|`readerName`|Read name.|null
+String| `subscriptionName`|Subscription name|None
```
Comment (315157973) [inline site2/docs/client-libraries-java.md:679]: The default value is not 0
When there is a single topic, the default subscription name is: "reader-" + 10-digit UUID
When there are multiple topics, the default subscription name is: "The default subscription name is: "reader-" + 10-digit UUID-" + 10-digit UUID

Comment (Jennifer88huang) [reply site2/docs/client-libraries-java.md]: thank you for your comment.

**Comment by Jennifer88huang** [comment]
> @Jennifer88huang thanks for your contribution. Need to add similar contents for C++ client as well?

Currently, we do not have any content related to Reader for C++ client. We can add it when improving C++ client as a whole. You can add a follow-up issue for it.

**Review by Anonymitaet — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

**Pull request merged**